src/CoolPropTools.cpp

```diff
@@ -17,7 +17,7 @@
 #include <stdint.h>
 #include <iostream>
 
-static double ftw_summer; // An evil global variable for the ftw function
+static thread_local unsigned long long ftw_summer; // An evil global variable for the ftw function
```

Comment:
+1 If you notice other things that are not thread-safe, PLEASE fix them.  Thread safeness is something I am only aware of in a limited fashion